bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeChecker.java

```diff
@@ -1188,7 +1203,12 @@ private static boolean checkIsLikeOnValue(Object sourceValue, BType sourceType,
             case TypeTags.XML_PI_TAG:
             case TypeTags.XML_TEXT_TAG:
                 if (sourceType.getTag() == TypeTags.XML_TAG) {
-                    return matchXMLType((XMLValue) sourceValue, targetType);
+                    return matchXMLTypeNonSequence((XMLValue) sourceValue, targetType);
```

Comment:
Shall we rename this to something like `checkIsLikeNonElementSingleton`

Reply:
Done